pkg/workspace/inference/preset-inferences.go

```diff
@@ -131,6 +133,17 @@ func CreatePresetInference(ctx context.Context, workspaceObj *v1beta1.Workspace,
 	}
 	if gpuConfig != nil {
 		skuNumGPUs = gpuConfig.GPUCount
+		// Calculate the minimum number of nodes required to satisfy the model's total GPU memory requirement.
+		// The goal is to maximize GPU utilization and not spread the model across too many nodes.
+		totalGPUMemoryRequired := resource.MustParse(inferenceParam.TotalGPUMemoryRequirement)
+		totalGPUMemoryPerNode := resource.NewScaledQuantity(int64(gpuConfig.GPUCount*gpuConfig.GPUMemGB), resource.Giga)
+		minimumNodes := 0
+		for ; totalGPUMemoryRequired.Sign() > 0; totalGPUMemoryRequired.Sub(*totalGPUMemoryPerNode) {
```

Comment:
I doubt the calculation algo here. For multinode inference, there are extra memory overhead for multi gpu communication. However, I can't give you a more accurate formula right now. Let's do some profiling later to check this overhead.